### ISSUE TITLE
Render problem: <WRAP col2 >

### ISSUE THREAD

**Issue by coastgnu** [LotarProject/dokuwiki-template-bootstrap3#108]
Moin Moin and Ciao Giuseppe

# &lt;WRAP col2&gt; doesn't work
For les than a second 2 cols are displayer.
Then only a single col shows up but at least it's 50% wide.

This is in both view modes in fluid container (100% width) and in normal mode view.

## Configuration of bootstrap3 
From <code>grep bootstrap3 conf/local.php</code>
```php
conf/local.php:$conf['template'] = 'bootstrap3';
conf/local.php:$conf['tpl']['bootstrap3']['bootstrapTheme'] = 'bootswatch';
conf/local.php:$conf['tpl']['bootstrap3']['fixedTopNavbar'] = '1';
conf/local.php:$conf['tpl']['bootstrap3']['showTools'] = 'logged';
conf/local.php:$conf['tpl']['bootstrap3']['hideLoginLink'] = '1';
conf/local.php:$conf['tpl']['bootstrap3']['individualTools'] = '1';
conf/local.php:$conf['tpl']['bootstrap3']['showIndividualTool'] = 'site,page';
conf/local.php:$conf['tpl']['bootstrap3']['showAdminMenu'] = '1';
conf/local.php:$conf['tpl']['bootstrap3']['semantic'] = '1';
conf/local.php:$conf['tpl']['bootstrap3']['schemaOrgType'] = 'NewsArticle';
conf/local.php:$conf['tpl']['bootstrap3']['fluidContainer'] = '1';
conf/local.php:$conf['tpl']['bootstrap3']['fluidContainerBtn'] = '1';
conf/local.php:$conf['tpl']['bootstrap3']['showPageTools'] = 'logged';
conf/local.php:$conf['tpl']['bootstrap3']['showPageId'] = '0';
conf/local.php:$conf['tpl']['bootstrap3']['showPageInfo'] = '0';
conf/local.php:$conf['tpl']['bootstrap3']['showBadges'] = '0';
conf/local.php:$conf['tpl']['bootstrap3']['showLoginOnFooter'] = '1';
conf/local.php:$conf['tpl']['bootstrap3']['tagsOnTop'] = '0';
```

## Configuration of other plugins 

From <code>grep  'plugin' conf/local.php</code>

```php
 * Auto-generated by config plugin
$conf['plugin']['clearhistory']['autoclearenabled'] = 1;
$conf['plugin']['clearhistory']['autoclearonlysmall'] = 1;
$conf['plugin']['odt2dw']['debugLvl'] = '0';
$conf['plugin']['odt2dw']['parserMimeTypeAuthorized'] = 'application/vnd.oasis.opendocument.text
$conf['plugin']['addnewpage']['addpage_exclude'] = 'wiki;playground;templates';
$conf['plugin']['addnewpage']['addpage_showroot'] = 0;
$conf['plugin']['addnewpage']['addpage_hideACL'] = 1;
$conf['plugin']['denyactions']['denyactions'] = 'index, media, recent, search, export_raw, export_xhtml, export_xhtmlbody, diff, backlink';
$conf['plugin']['pagelist']['showcomments'] = 1;
$conf['plugin']['pagelist']['showtags'] = 1;
$conf['plugin']['indexmenu']['only_admins'] = 1;
$conf['plugin']['indexmenu']['headpage'] = ':same:,:inside:';
$conf['plugin']['pluginrepo']['db_name'] = 'sqlite:/var/www/gnuwhv/data/meta/data.sqlite3';
$conf['plugin']['pluginrepo']['releases'] = '2014-09-29 "Hrun" , 2014-05-05 "Ponder Stibbons", 2013-12-08 "Binky", 2013-05-10 "Weatherwax", 2012-10-13 "Adora Belle", 2011-11-10 “Angua”, 2011-05-25 "Rincewind", 2010-11-07 "Anteater", 2009-12-25 "Lemmings", 2009-02-14, 2008-05-05, 2007-06-26';
$conf['plugin']['tag']['toolbar_icon'] = 1;
$conf['plugin']['tag']['list_tags_of_subns'] = 1;
$conf['plugin']['dw2pdf']['usestyles'] = 'pagequery,pagelist,inlinetoc';
$conf['plugin']['latex']['xsize_limit'] = 20000;
$conf['plugin']['latex']['ysize_limit'] = 10000;
$conf['plugin']['latex']['string_length_limit'] = 20000;
$conf['plugin']['latex']['latex_path'] = '/usr/bin/latex';
$conf['plugin']['latex']['dvips_path'] = '/usr/bin/dvips';
$conf['plugin']['latex']['identify_path'] = '/usr/bin/identify ';
$conf['plugin']['latex']['convert_path'] = '/usr/bin/convert';
$conf['plugin']['latex']['convert_options'] = '-density 1200 -trim -transparent "#FFFFFF" -units pt -pointsize 16 ';
$conf['plugin']['backup']['filterbackups'] = 1;
$conf['plugin']['gallery']['thumbnail_width'] = 200;
$conf['plugin']['gallery']['thumbnail_height'] = 145;
$conf['plugin']['gallery']['image_width'] = 1640;
$conf['plugin']['gallery']['image_height'] = 1080;
$conf['plugin']['gallery']['cols'] = 4;
$conf['plugin']['gallery']['options'] = 'lightbox,showtitle';
$conf['plugin']['latexit']['font_size'] = 12;
$conf['plugin']['latexit']['document_lang'] = 'german';
$conf['plugin']['semantic']['defaultSchemaOrgType'] = 'Recipe';
$conf['plugin']['svg']['output'] = 'embed';
$conf['plugin']['discussion']['allowguests'] = '0';
$conf['plugin']['discussion']['showguests'] = '0';
$conf['plugin']['discussion']['useavatar'] = '0';
$conf['plugin']['discussion']['userealname'] = '0';
$conf['plugin']['discussion']['visibilityButton'] = '1';
$conf['plugin']['discussion']['moderatorgroups'] = 'talkmaster';
$conf['plugin']['whoisonline']['displayline'] = '<a href="/users:{username}">{fullname}</a>  - <a href="/{pageid}">{pageid}</a>  - <a href="/{url}">{url}</a> ';
$conf['plugin']['bookcreator']['toolbar'] = 'never';
$conf['plugin']['definitionlist']['dt_fancy'] = '1';
$conf['plugin']['syntaxhighlighter3']['theme'] = 'shThemeEclipse.css';
$conf['plugin']['userpagecreate']['target'] = 'userpages';
$conf['plugin']['userpagecreate']['template'] = '__template.txt';
$conf['plugin']['userpagecreate']['create_summary'] = 'Automatisch erstellte Anwender-Seite ';
$conf['plugin']['translation']['translations'] = 'en nl nds';
```
## Activated Plugins
Listed with Popularity Contest Plugin: <code>?do=admin&page=popularity</code>
```php
plugin	acl
plugin	addnewpage
plugin	backup
plugin	bootswrapper
plugin	bureaucracy
plugin	clearhistory
plugin	codebutton
plugin	codeprettify
plugin	comment
plugin	config
plugin	data
plugin	datatables
plugin	datatemplate
plugin	denyactions
plugin	dw2pdf
plugin	edittable
plugin	extension
plugin	folded
plugin	gallery
plugin	hidden
plugin	hiddenSwitch
plugin	ifauth
plugin	imagebox
plugin	include
plugin	indexmenu
plugin	info
plugin	inlinetoc
plugin	keyboard
plugin	latex
plugin	latexit
plugin	log
plugin	loglog
plugin	nspages
plugin	odt
plugin	odt2dw
plugin	pagelist
plugin	pagemod
plugin	pagequery
plugin	pageredirect
plugin	plugin
plugin	pluginrepo
plugin	popularity
plugin	popupviewer
plugin	querychangelog
plugin	redirect
plugin	reflect
plugin	revert
plugin	safefnrecode
plugin	searchtablejs
plugin	semantic
plugin	sortablejs
plugin	sqlite
plugin	statistics
plugin	styling
plugin	superacl2
plugin	svgpureinsert
plugin	tablewidth
plugin	tag
plugin	text
plugin	togglewrap
plugin	toolbox
plugin	toolbuttondel
plugin	upgrade
plugin	usermanager
plugin	wrap
plugin	yalist

```

Please let me know if you need any further information to trace this odd behaviour.

regards,
Thomas

**Comment by LotarProject** [comment]
Hi! Please send a snippet of code for test.

At your disposal,
Joseph

**Comment by coastgnu** [comment]
====== test  ======
First 
===== long lists causing problems  =====
Refresh the rendering, F5 or Ctrl+R, in firefox to reproduce the cols to show up for 1 second and dissapear.

I first saw this in an indexmenu.  A standup comedy like two step forward, one step back appearance.

Has a bit of Laurel and Hardy, "Sons of The Desert." I never before recognized such a behaviour in a Dokuwiki.

<WRAP col2>
{{indexmenu>:people#3}}

plugin:Indexmenu, listing the namespace people which goes from a to z (19 subnamespaces) and has 86 people.

</WRAP>

<WRAP col2>
  * 1
  * 2
  * 3
  * 4
  * 5
  * 6
  * 7
  * 8
  * 9
  * 0
  * 1
  * 2
  * 3
  * 4
  * 5
  * 6
  * 7
  * 8
  * 9
  * 0
  * 1
  * 2
  * 3
  * 4
  * 5
  * 6
  * 7
  * 8
  * 9
  * 0
  * 1
  * 2
  * 3
  * 4
  * 5
  * 6
  * 7
  * 8
  * 9
  * 0
</WRAP>

===== paragraphs don't cause problems =====

<WRAP col2>
1

2

3

4

5

6

7

8

9

0

1

2

3

4

5

6

7

8

9

0

1

2

3

4

5

6

7

8

9

0

1

2

3

4

5

6

7

8

9

0

</WRAP>

===== Header, paragraphs and short lists don't cause problems =====
<WRAP col2>
==== Headline ====

1

2

3
=== Headline ===
4
  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
5

6
=== Headline ===
7

8

9
== Headline ==
0
==== Headline ====
1

2

3
=== Headline ===
4
  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
5

6
=== Headline ===
7

8

9
== Headline ==
0
==== Headline ====
1

2

3
=== Headline ===
4
  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
5

6
=== Headline ===
7

8

9
== Headline ==
0
==== Headline ====
1

2

3
=== Headline ===
4
  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
5

6
=== Headline ===
7

8

9
== Headline ==
0

</WRAP>


===== header, paragraphs and ong lists start causing problems =====
Compared to dokuwiki's default template the cols begin to split wrong.
<WRAP col2>
==== Headline ====

01

02

03
=== Headline ===
04
  * a
    * aa
    * ab
    * ac
    * ad
    * ae
    * af
  * b
    * ba
    * bb
    * bc
    * bd
    * be
    * bf
  * c
    * ca
    * cb
    * cc
    * cd
    * ce
    * cf
  * d
    * da
    * db
    * dc
    * dd
    * de
    * df
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
05

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
06
=== Headline ===
07

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
08

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
09
== Headline ==
10
==== Headline ====
11

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
12

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
13
=== Headline ===
14
  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
15

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
16
=== Headline ===
17

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
18

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
19
== Headline ==
20
==== Headline ====
21

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
22

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
23
=== Headline ===
24
  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
25

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
26
=== Headline ===
27

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
28

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
29
== Headline ==
30
==== Headline ====
31

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
32

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
33
=== Headline ===
34
  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
35

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef
36
=== Headline ===
37

  * a
  * b
  * c
  * d
  * e
    * ea
    * eb
    * ec
    * ed
    * ef

</WRAP>


<WRAP col2>

===== header, paragraphs and ong lists start causing problems =====

==== header ====
a lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum 
  * lorem ipsum lorem ipsum 
  * lorem ipsum lorem ipsum 
  * lorem ipsum lorem ipsum 
  * lorem ipsum lorem ipsum 
    * lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum 
    * lorem ipsum lorem ipsum lorem ipsum 
      * lorem ipsum lorem ipsum lorem ipsum 
    * lorem ipsum lorem ipsum lorem lorem ipsum 
  * lorem ipsum lorem ipsum 
  * lorem ipsum lorem ipsum 
  * lorem ipsum lorem ipsum 
  * lorem ipsum lorem ipsum 
</WRAP>

The list don't split but it gets split in dokuwiki's default template.

And, the edit button opens an edit window over the last h2 sections, which is a bit irritating, I wouldn't call this a bug. But it may help to trace this odd behaviour.

regards,
thomas

**Comment by LotarProject** [comment]
Hi! I'have tested your code in differents DokuWiki environment with ``master`` and ``develop`` branch and don't see the issue.

Can you fetch the last template release ?

At your disposal,
Joseph

**Comment by coastgnu** [comment]
Pardon me for the delay...

Strange, it works again and i've no clue why.

Since 2015-11-05 there where a couple of updates, mostly plugins. So it might have been a conflict with an other plugin i would suppose.


Sorry for the noise,
thomas

**Comment by selfthinker** [comment]
Someone just posted the [same issue in the German DokuWiki sub forum](https://forum.dokuwiki.org/thread/13628). With the simple example of
```
<WRAP col2>
  * list item 1
  * list item 2
  * list item 3
  * list item 4
</WRAP>
```
I can reproduce it in Firefox and IE Edge but not in Chrome.

This is happening due to d97e66d9895937628e9ef55e2a4848be7df699dc which was trying to fix #55. If you read my comment in the old DokuWiki bug tracker linked in #55, you will see that we were aware that either solution would cause lots of issues as the same happened in the standard template, which was the reason why we removed the solution again.
